### PR TITLE
implementing hvac drivers

### PR DESCRIPTION
**Pull request by hendrikVandeput**

### CODE REVIEW

File: src/gateway/thermostat/gateway/hvac_drivers.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+class HvacDriverParent(object):
+    def __init__(self, hvac_id):
+        self._hvac_id      = hvac_id  # TODO: currently hvac_id is the id of the thermostat group
+        self._old_mode     = None  # remember previous state to not spam the output with changes
+        self._old_delta_t  = None  # remember previous state to not spam the output with changes
+        self._old_setpoint = None  # remember previous state to not spam the output with changes
+
+
+    def steer(self, mode, delta_t=None, setpoint=None):  # type: (self, string, float, float) -> None
+    	if mode is None:
```
[the remark this answers is not in the document]
Comment (hendrikVandeput) [reply src/gateway/thermostat/gateway/hvac_drivers.py:17]: delta_t and setpoint will become relevant in the future when we add extra drivers, but are not at this moment in time. To accommodate the transition, I've made it nullable.

File: src/gateway/dto/hvac_driver.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+if False:  # MYPY
+    from typing import Any, Optional
+
+
+class HvacContactDTO(BaseDTO):
+    def __init__(self, output_id, mode, value, type):
+        self.output_id = output_id  # type: int
+        self.mode = mode  # type: int
+        self.value = value  # type: int
+        self.type = type  # type: int
```
Comment (hendrikVandeput) [inline src/gateway/dto/hvac_driver.py:12]: string

File: src/gateway/models.py [hunk 3]
```diff
@@ -380,31 +382,54 @@ class Modes(object):
...
+                                               primaryjoin='and_(ThermostatGroup.id == HvacOutputLink.hvac_id, HvacOutputLink.mode == "cooling")',
+                                               order_by='asc(HvacOutputLink.id)',
+                                               back_populates='hvac')  # type: RelationshipProperty[List[HvacOutputLink]]
+
+
+
+
+# thermostatsV2
+class HvacOutputLink(Base):
+    MODES = [
```
Comment (hendrikVandeput) [inline src/gateway/models.py:401]: step away from choicetypes -> make dicts

File: src/gateway/models.py [hunk 4]
```diff
@@ -380,31 +382,54 @@ class Modes(object):
...
+        (u'heating', u'heating'),
+        (u'cooling', u'cooling'),
+        (u'off', u'off')
+    ]
+    TYPES = [
```
Comment (hendrikVandeput) [inline src/gateway/models.py:406]: can be removed

File: src/gateway/thermostat/gateway/hvac_drivers.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                    mode      = hvac_output_link.mode,
+                    value     = hvac_output_link.value,
+                    type      = hvac_output_link.type
+                )
+            new_output_list.append(new_output)
+        self._output_list = new_output_list
+
+
+    # change value of outputs according to setting fetched from database
+    def steer(self, mode, delta_t=None, setpoint=None):  # type: (str, Optional[float], Optional[float]) -> None
```
Comment (hendrikVandeput) [inline src/gateway/thermostat/gateway/hvac_drivers.py:65]: ook toevoegen aan de parent om duidelijk te maken dat deze functie nodig is om aan te passen
met status not implemented

**Review by LnL7 — APPROVED** [review]

**Review by LnL7 — APPROVED** [review]

**Pull request merged**